### DISCUSSION TITLE
从v4升级到v5失败，碰到好诡异的问题

### DISCUSSION THREAD

**Discussion by Chasen-Zhang** [ant-design/ant-design#50432] [Q&A]
首先我把cssVar 打开
![image](https://github.com/user-attachments/assets/e9631837-8e36-4817-ad62-8467d54d8b48)

下面这个图是我基于antd5 Button 组件增强的组件，这个组件在另外一个UI库（称为A库），然后业务项目（B库）也是基于antd5 做的，B库引入A的组件，发现样式丢了，
![image](https://github.com/user-attachments/assets/fdbaa0ab-74e6-41a2-ac3b-f9c5d9a3e07f)

但是我要是打开其他页面有按钮的（B库直接引入的antd5 Button组件），然后回退到这个页面，上图的图片的按钮会正常，丢失的.ant-btn 样式会正常出现（但是一刷新又没了），
![image](https://github.com/user-attachments/assets/82463f31-0197-4c22-879b-fd5f8366aeb8)

这个是什么原因呢，

这个虽然正常出现了，但是某些--ant-xxx 变量还是灰色未定义的 @Wxh16144

**Comment by Wxh16144** [comment]
我猜测可能是样式插入顺序问题🤔

**Reply by Chasen-Zhang** [reply]
我的UI库是用你们家另一个开源项目dumi+antd5 做的，大佬有什么思路么？ @Wxh16144

**Comment by Chasen-Zhang** [comment]
我发现从其他基于antd5 做的组件库引入的组件都是这样，样式丢失，--ant--xx-xx这些变量也没定义 @Wxh16144

**Comment by Chasen-Zhang** [comment]
我的UI库（A库）本地预览这些变量都是有作用到的
![image](https://github.com/user-attachments/assets/48019c75-703e-45c1-ae75-b1c0f6b47a77)

所以我的业务项目是要做什么特殊的配置吗
